src/Time/TimeSteppers/RungeKutta3.hpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class RungeKutta3.
+
+#pragma once
+
+#include <deque>
+#include <string>
+
+#include "ErrorHandling/Error.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeSteppers/TimeStepper.hpp"
+
+namespace TimeSteppers {
+
+/// A "strong stability-preserving" 3rd-order Runge-Kutta time-stepper.
+/// Major reference:  J. Hesthaven & T. Warburton, Nodal Discontinuous
+/// Galerkin Methods. section 5.7
+class RungeKutta3 : public TimeStepper::Inherit {
+ public:
+  //using OptionsList = tmpl::list<>;
+  //static std::string class_id() { return "RungeKutta3"; }
+  //static constexpr OptionString_t help = {
+  //    "A third-order strong stability-preserving Runge-Kutta time-stepper."};
+
+  RungeKutta3() = default;
+  RungeKutta3(const RungeKutta3&) = default;
+  RungeKutta3& operator=(const RungeKutta3&) = default;
+  RungeKutta3(RungeKutta3&&) = default;
+  RungeKutta3& operator=(RungeKutta3&&) = default;
+  ~RungeKutta3() override = default;
+
+  template <typename Vars, typename DerivVars>
+  TimeDelta update_u(
+      Vars& u,
+      const std::deque<std::tuple<Time, Vars, DerivVars>>& history,
+      const TimeDelta& time_step) const;
+
+  size_t number_of_substeps() const noexcept override;
+
+  size_t number_of_past_steps() const noexcept override;
+
+  bool is_self_starting() const noexcept override;
+
+  double stable_step() const noexcept override;
+};
+
+template <typename Vars, typename DerivVars>
+TimeDelta RungeKutta3::update_u(
+    Vars& u,
+    const std::deque<std::tuple<Time, Vars, DerivVars>>& history,
+    const TimeDelta& step_size) const {
+  const size_t substep = history.size() - 1;
+  const auto& vars = std::get<1>(history.back());
+  const auto& dt_vars = std::get<2>(history.back());
+  const auto& U0 = std::get<1>(history[0]);
+
+  switch (substep) {
+    case 0: {
+      // from (5.32) of Hesthaven
```

Comment:
In this case I don't see what it would bind to, so no. As I see it, we have three options: 
1) leave as is
2) move all this type of documentation onto the function giving a nice LaTeX documentation explaining how the method works.
3) Same as two but put the Doxygen comment on the class

I don't have a strong preference here yet, but will think about it more. I'm mostly inclined to vote for requiring 3) but not denying 1) since there is merit for documentation in code. Having said that, Google's guide is that algorithm documenting documentation should be before the tricky bits of code, not dispersed throughout. Since I think here the documentation will be more likely to be read if on the class, I think that would be the most logical place. Just my current thoughts on the manner, let's have a discussion here about it.